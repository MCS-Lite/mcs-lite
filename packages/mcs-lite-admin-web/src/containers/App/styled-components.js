import styled from 'styled-components';

export const HeightContainer = styled.div`
  height: 100%;
`;

export const Fixed = styled.div`
  position: fixed;
  bottom: 20px;
  width: 100%;
`;

export const ToastContainer = styled.div`
  padding: 0 100px;

  > * {
    margin-top: 8px;
  }
`;
