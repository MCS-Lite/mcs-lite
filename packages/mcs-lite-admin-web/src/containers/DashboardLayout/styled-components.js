import styled from 'styled-components';
import P from 'mcs-lite-ui/lib/P';
import Logo from '../../components/Logo';

export const Container = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
`;

export const Header = styled.header`
  flex-basis: 40px;
  flex-shrink: 0;
  box-sizing: border-box;
  background-color: ${props => props.theme.color.grayLight};
  border-bottom: 1px solid ${props => props.theme.base.bodyBackgroundColor};
  padding: 0 20px;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const HeaderItemWrapper = styled.div`
  display: flex;
  height: 100%;

  > * {
    border-left: 1px solid ${props => props.theme.base.bodyBackgroundColor};

    &:last-child {
      border-right: 1px solid ${props => props.theme.base.bodyBackgroundColor};
    }
  }
`;

export const HeaderItem = styled.div`
  height: 100%;
  display: flex;
  padding: 0 15px;
  align-items: center;
  cursor: pointer;
  color: ${props => props.theme.color.primary};

  > ${P} {
    margin-left: 5px;
  }
`;

export const StyledLogo = styled(Logo)`
  height: 22px;
  padding: 9px 0;
`;

export const Body = styled.div`
  flex-grow: 1;
  display: flex;

  ${'' /* Remind: fixed for firefox: https://goo.gl/Rjqio3 */}
  min-height: 0;
`;

export const Nav = styled.nav`
  flex-basis: 150px;
  box-sizing: border-box;
  background-color: ${props => props.theme.color.grayLight};
  border-right: 1px solid ${props => props.theme.base.bodyBackgroundColor};
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

export const NavItem = styled.div`
  width: : 100%;
  height: ${props => props.theme.height.normal};
  display: flex;
  padding: 0 20px;
  align-items: center;
  cursor: pointer;
`;

export const NavItemControl = styled(NavItem)`
  color: ${props => props.theme.color.primary};
  border-top: 1px solid ${props => props.theme.base.bodyBackgroundColor};

  > ${P} {
    margin-left: 5px;
  }
`;

export const Main = styled.main`
  flex-grow: 1;
  background-color: ${props => props.theme.color.white};
  overflow: auto;
`;
