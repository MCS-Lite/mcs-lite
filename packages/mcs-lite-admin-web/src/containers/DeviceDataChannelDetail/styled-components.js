import styled from 'styled-components';
import { MobileContentWrapper, Small, A } from 'mcs-lite-ui';

export const CardContainer = styled(MobileContentWrapper)`
  padding: 8px 8px 0 8px;
`;

export const StyledSamll = styled(Small)`
  display: block;
  margin-top: 4px;
  color: ${props => props.theme.color.grayBase};
`;

export const HistoryContainer = styled(MobileContentWrapper)`
  padding: 16px;
`;

export const ChartWrapper = styled.div`
  height: 300px;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const HistoryHeader = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  margin-bottom: 8px;
`;

export const ResetWrapper = styled(A)`
  display: flex;
  font-size: 24px;
  align-items: center;

  > *:first-child {
    margin-right: 4px;
  }
`;
