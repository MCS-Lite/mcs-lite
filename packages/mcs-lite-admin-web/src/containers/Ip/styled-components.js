import styled from 'styled-components';
import Button from 'mcs-lite-ui/lib/Button';

export const StyledButton = styled(Button)`
  margin-top: 10px;
`;

export const NaWrapper = styled.div`
  margin-top: 40px;
  text-align: center;
`;
