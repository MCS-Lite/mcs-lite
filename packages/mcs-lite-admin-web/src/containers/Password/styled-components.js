import styled from 'styled-components';
import { P, MobileContentWrapper } from 'mcs-lite-ui';

export const Container = styled(MobileContentWrapper)`
  padding: 16px 16px 56px 16px;

  > div + div {
    margin-top: 8px;
  }

  > div:last-child {
    margin-bottom: 16px;
  }
`;

export const Label = styled.label`
  display: block;
  margin-bottom: 4px;
`;

export const ButtonWrapper = styled.div`
  display: flex;
  padding: 8px 16px;

  > * {
    flex: 1;
  }

  > *:first-child {
    margin-right: 8px;
  }
`;

export const StyledP = styled(P)`
  margin-top: 4px;
`;
