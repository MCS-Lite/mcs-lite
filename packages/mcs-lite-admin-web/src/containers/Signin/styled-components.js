import styled from 'styled-components';
import Hr from 'mcs-lite-ui/lib/Hr';
import P from 'mcs-lite-ui/lib/P';
import Logo from '../../components/Logo';

export const StyledLogo = styled(Logo)`
  width: 200px;
`;

export const ErrorMessage = styled(P)`
  text-align: center;

  &:not(:empty) {
    margin-top: 20px;
  }
`;

export const StyledHr = styled(Hr)`
  margin-top: 20px;
  margin-bottom: 10px;
`;

export const Layout = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  height: 100%;
  width: 260px;
  margin: 0 auto;
`;

export const Form = styled.form`
  > input + input {
    margin-top: 10px;
  }

  > input:last-child {
    margin-top: 40px;
  }
`;
