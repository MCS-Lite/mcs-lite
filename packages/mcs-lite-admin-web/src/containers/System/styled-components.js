import styled from 'styled-components';
import Button from 'mcs-lite-ui/lib/Button';
import P from 'mcs-lite-ui/lib/P';
import CodeMirror from 'react-codemirror';

export const StyledButton = styled(Button)`
  margin-top: 10px;
`;

export const TabWrapper = styled.div`
  display: inline-block;
  border-bottom: 1px solid ${props => props.theme.color.grayDark};
  margin-bottom: 10px;
`;

export const Message = styled(P)`
  margin-top: 5px;
`;

export const StyledCodeMirror = styled(CodeMirror)`
  > .CodeMirror{
    border: 1px solid ${props => (props.error ? props.theme.color.error : props.theme.color.grayDark)};
    border-radius: 3px;
    height: 320px;
    color: ${props => props.theme.color.black};
  }

  .CodeMirror-linenumber {
    text-align: center;
    color: ${props => props.theme.color.grayDark};
    background-color: ${props => props.theme.color.grayLight};
  }

  .CodeMirror-line {
    padding-left: 10px;
  }
`;
