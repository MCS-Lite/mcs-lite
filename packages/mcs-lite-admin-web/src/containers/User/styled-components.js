import styled from 'styled-components';
import CommonDialog from 'mcs-lite-ui/lib/Dialog/CommonDialog';
import InputGroup from 'mcs-lite-ui/lib/InputGroup';
import A from 'mcs-lite-ui/lib/A';
import P from 'mcs-lite-ui/lib/P';

export const InputFilterWrapper = styled.div`
  display: flex;
  align-items: center;

  > ${InputGroup} {
    flex-grow: 1;
  }

  > ${A} {
    margin-left: 10px;
  }
`;

export const FooterWrapper = styled.div`
  position: fixed;
  bottom: 0;
  padding-top: 10px;
  padding-bottom: 20px;
  width: 100%;
  background-color: ${props => props.theme.color.white};
  line-height: 0;

  > ${A} {
    display: inline-flex;
    align-items: center;

    div {
      margin-left: 5px;
      line-height: 1em;
    }
  }
`;

export const StyledCommonDialog = styled(CommonDialog)`
  padding-top: 40px;

  main {
    align-items: flex-start;
  }
`;

export const TabWrapper = styled.div`
  border-bottom: 1px solid ${props => props.theme.color.grayBase};
  margin-bottom: 10px;
`;

export const RadioWrapper = styled.div`
  display: flex;
  align-items: center;

  > *:first-child {
    margin-right: 5px;
  }

  label {
    margin-right: 10px;
  }

  input {
    margin-right: 5px;
  }
`;

export const ErrorMessageP = styled(P)`
  padding-top: 5px;
`;

export const InputFileWrapper = styled.div`
  position: relative;
  height: ${props => props.theme.height.normal};

  > * {
    position: absolute;
    width: 100%;
    pointer-events: none;
  }

  input[type='file'] {
    opacity: 0;
    cursor: pointer;
    pointer-events: all;
  }
`;
