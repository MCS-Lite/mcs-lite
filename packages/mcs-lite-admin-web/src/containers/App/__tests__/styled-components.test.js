import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ThemeProvider } from 'styled-components';
import { theme } from 'mcs-lite-theme';
import { HeightContainer, Fixed, ToastContainer } from '../styled-components';

jest.mock('mcs-lite-ui');

it('should render components correctly', () => {
  const wrapper = mount(
    <ThemeProvider theme={theme}>
      <div>
        <HeightContainer />
        <Fixed />
        <ToastContainer />
      </div>
    </ThemeProvider>,
  );

  expect(toJson(wrapper.find('div'))).toMatchSnapshot();
});
