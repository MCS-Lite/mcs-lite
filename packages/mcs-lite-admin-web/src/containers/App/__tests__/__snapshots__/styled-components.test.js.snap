// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render components correctly 1`] = `
Array [
  <div>
    <styled.div>
        <div
            className="sc-bdVaJa dGIfxd"
        />
    </styled.div>
    <styled.div>
        <div
            className="sc-bwzfXH fJgWw"
        />
    </styled.div>
    <styled.div>
        <div
            className="sc-htpNat ctRuvZ"
        />
    </styled.div>
</div>,
  <div
    className="sc-bdVaJa dGIfxd"
/>,
  <div
    className="sc-bwzfXH fJgWw"
/>,
  <div
    className="sc-htpNat ctRuvZ"
/>,
]
`;
